deployment/terraform/create.go

```diff
@@ -324,6 +332,175 @@ func (t *Terraform) Create(initData bool) error {
+func (t *Terraform) checkCloudWatchLogsPolicy() error {
+	type polDocStmtPrincipal struct {
+		Service string `json:"Service"`
+	}
+	type polDocStmt struct {
+		Effect    string              `json:"Effect"`
+		Principal polDocStmtPrincipal `json:"Principal"`
+		Action    []string            `json:"Action"`
+		Resource  string              `json:"Resource"`
+	}
+	type polDoc struct {
+		Version   string       `json:"Version"`
+		Statement []polDocStmt `json:"Statement"`
+	}
+
+	// Create CloudWatchLogs client
+	cfg, err := awsconfig.LoadDefaultConfig(context.Background(),
+		awsconfig.WithSharedConfigProfile(t.config.AWSProfile),
+	)
+	if err != nil {
+		return err
+	}
+	cwclient := cloudwatchlogs.NewFromConfig(cfg)
+
+	// Function to check the policy we need in AWS
+	findPolicy := func(policies []cwtypes.ResourcePolicy) (cwtypes.ResourcePolicy, error) {
+		for _, policy := range policies {
+			var doc polDoc
+			err := json.Unmarshal([]byte(*policy.PolicyDocument), &doc)
+			if err != nil {
+				return cwtypes.ResourcePolicy{}, err
+			}
+
+			if len(doc.Statement) == 0 {
+				continue
+			}
+
+			for _, statement := range doc.Statement {
+				if statement.Effect != "Allow" {
+					continue
+				}
+
+				if statement.Principal.Service != "es.amazonaws.com" {
+					continue
+				}
+				if !slices.Contains(statement.Action, "logs:PutLogEventsBatch") {
+					continue
+				}
+
+				if !slices.Contains(statement.Action, "logs:PutLogEvents") {
+					continue
+				}
+
+				if !slices.Contains(statement.Action, "logs:CreateLogStream") {
+					continue
+				}
+
+				if statement.Resource != "arn:aws:logs:*" {
+					continue
+				}
+
+				return policy, nil
+			}
+
+		}
+
+		return cwtypes.ResourcePolicy{}, fmt.Errorf("policy not found")
+	}
+
+	// Iterate through all available policies and check if one of them passes
+	// the verification findPolicy check
+	var nextToken *string
+	limit := int32(10)
+	hasMore := true
+	found := false
+	var policy cwtypes.ResourcePolicy
+	for hasMore {
+		input := cloudwatchlogs.DescribeResourcePoliciesInput{
+			Limit:     &limit,
+			NextToken: nextToken,
+		}
+		output, err := cwclient.DescribeResourcePolicies(context.Background(), &input)
+		if err != nil {
+			return err
+		}
+		nextToken = output.NextToken
+		hasMore = nextToken != nil
+
+		policy, err = findPolicy(output.ResourcePolicies)
+		if err != nil {
+			continue
+		}
+
+		found = true
+		break
+	}
+
+	if found {
+		// There's at least one policy that has all permissions we need,
+		// so we can skip creating one
+		mlog.Debug("CloudWatchLogs policy found", mlog.String("name", *policy.PolicyName))
+		return nil
+	}
+
+	doc := polDoc{
+		Version: "2012-10-17",
+		Statement: []polDocStmt{
+			{
+				Effect: "Allow",
+				Principal: polDocStmtPrincipal{
+					Service: "es.amazonaws.com",
+				},
+				Action: []string{
+					"logs:PutLogEventsBatch",
+					"logs:PutLogEvents",
+					"logs:CreateLogStream",
```

Comment:
Maybe we can store them as constants? Since they are also being used in the comparison function.